### PR TITLE
Accessibility: fixing DataGridView BoundingRectangle property for row, data cells and header cells

### PR DESCRIPTION
**Pull request by M-Lipin**
The change is related to the issue #710.

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewCell.cs [hunk 1]
```diff
@@ -5134,10 +5139,34 @@ internal Rectangle GetAccessibleObjectBounds(AccessibleObject parentAccObject)
                     columnRect.Width = this.owner.OwningColumn.Width;
                 }
 
-                rowRect.X = columnRect.X;
-                rowRect.Width = columnRect.Width;
+                var cellRight = columnRect.Left + columnRect.Width;
```
Comment (M-Lipin) [inline src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewCell.cs]: Does the change take into account right-to-left environment? Please update the code accordingly if need.

Comment (vladimir-krestov) [reply src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewCell.cs]: Fixed this in https://github.com/dotnet/winforms/pull/711/commits/23cd06af0bab4adb0f33b5b02e3be35023624e16

**Comment by vladimir-krestov** [comment]
- RTL: fixed the issue with incorrect BoundingRectangle property of row headers in right-to-left environment.
- Refactoring: mainly removed buggy, performance inefficient and recursive implementation of accessible bounds calculation.

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewCell.cs [hunk 2]
```diff
@@ -1508,7 +1510,7 @@ internal static void FormatPlainTextAsHtml(string s, TextWriter output)
                         break;
                     // 
                     default:
-                        #if ENTITY_ENCODE_HIGH_ASCII_CHARS
+#if ENTITY_ENCODE_HIGH_ASCII_CHARS
```
[the remark this answers is not in the document]
Comment (vladimir-krestov) [reply src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewCell.cs]: I removed this unused managed define and condition. Could you please confirm that the change is correct?

**Pull request merged**